### ISSUE TITLE
Allow projects to opt for more frequent builds (based on availability)

### ISSUE THREAD

**Issue by inferno-chromium** [google/oss-fuzz#2584]
Currently, we build once a day. This does not work well for many projects.

**Comment by jonathanmetzman** [comment]
One idea to optimize this: Only do more frequent builds for ASAN/libFuzzer.

**Comment by Dor1s** [comment]
We can customize triggers in https://github.com/google/oss-fuzz/blob/master/infra/gcb/jenkins_config/base_job.xml while processing projects in https://github.com/google/oss-fuzz/blob/master/infra/gcb/sync.py (can rely on some flag in the project.yaml file, though I think we didn't want to add more flags to it).

It should work, but will not be efficient, as we always do a fresh build and changing this will be a bigger effort (and possibly a step into the land of instability). So, if we want to enable more frequent builds for a few projects, we probably shouldn't worry about this.

A super fancy way to decide how often we should build a project is to check its revision control log, get an approximate number of commits per day, and adjust frequency based on that. For example, if a project has 3 or less commits per day, we build it only once. If there are 4-6 builds a day, we build two times a day, and so on. Concrete thresholds can be different, of course :)

**Comment by oliverchang** [comment]
I think if we make changes to the way we trigger builds, we should take the chance to get rid of Jenkins. It's unnecessary at this point, as we are just relying on it to schedule builds.

Perhaps we can have something simple based on [cloud scheduler](https://cloud.google.com/scheduler/) which creates builds on cloud build directly.

**Comment by evverx** [comment]
Since this topic came up in https://github.com/google/oss-fuzz/issues/2525#issuecomment-502953374 relatively recently, I though I'd chime in. I think more frequent builds will help to catch issues earlier but this will be helpful only if maintainers pay attention to whatever happens on monorail (which as far as I understand is kind of foreign to some (probably most) people looking at the reports in the sense that their workflows don't revolve around it at all). My guesstimate for systemd would be that it takes about a week for someone to just notice a new issue reported by OSS-Fuzz on monorail so it doesn't really matter whether the code is going to be build once a day or, say, three times a day. To me, in terms of integrating OSS-Fuzz into the development workflow, it would be much more useful if oss-fuzz provided an API I could use to automatically mirror bugs to places where they are really looked into. But I digress.
